### ISSUE TITLE
aw-watcher-window, aw-watcher-afk process crashes

### ISSUE THREAD

**Issue by 200928** [ActivityWatch/activitywatch#938]
- [x] I am on the [latest](https://github.com/ActivityWatch/activitywatch/releases/latest) ActivityWatch version.
 - [x] I have searched the issues of this repo and believe that this is not a duplicate.

- **OS name and version**: 
   - Linux Arch 6.4.7-arch1-1, WM: dwm
- **ActivityWatch version**: 
   - 0.12.2-1 
## Describe the bug

I created the `start.sh` script using the method above the [URL.](https://docs.activitywatch.net/en/latest/running-on-gnome.html#running-on-gnome) When I run the script in the terminal, the aw-watcher-window, aw-watcher-afk and aw-server can run normally in the first few tens of seconds. 
After a while the following error occurs：
```shell
% ./start.sh                                                                                                             :( 130 23-08-19 - 9:56:16
2023-08-19 09:56:17 [INFO ]: aw-watcher-window started  (aw_watcher_window.main:62)
2023-08-19 09:56:17 [INFO ]: aw-watcher-afk started  (aw_watcher_afk.afk:60)
2023-08-19 09:56:17 [INFO ]: Using storage method: peewee  (aw_server.main:33)
2023-08-19 09:56:17 [INFO ]: Starting up...  (aw_server.main:41)
2023-08-19 09:56:17 [INFO ]: Using database file: /home/minghui/.local/share/activitywatch/aw-server/peewee-sqlite.v2.db  (aw_datastore.storages.peewee:126)
 * Serving Flask app 'aw-server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
2023-08-19 09:56:17 [INFO ]:  * Running on http://localhost:5600 (Press CTRL+C to quit)  (werkzeug:224)
2023-08-19 09:56:18 [INFO ]: Connection to aw-server established by aw-watcher-window  (aw_client.client:436)
2023-08-19 09:56:18 [INFO ]: Connection to aw-server established by aw-watcher-afk  (aw_client.client:436)
2023-08-19 09:56:21 [INFO ]: Received heartbeat after pulse window, inserting as new event. (bucket: aw-watcher-window_Arch)  (aw_server.api:279)
2023-08-19 09:56:28 [INFO ]: Received heartbeat after pulse window, inserting as new event. (bucket: aw-watcher-afk_Arch)  (aw_server.api:279)
为 org.freedesktop.Notifications 调用 StartServiceByName 时出错：已到超时限制
% 2023-08-19 09:57:17 [INFO ]: window-watcher stopped because parent process died  (aw_watcher_window.main:101)            :( 1 23-08-19 - 9:57:17
2023-08-19 09:57:18 [INFO ]: afkwatcher stopped because parent process died  (aw_watcher_afk.afk:80)
``` 
I don't know where the problem is, hope to get a solution.

**Comment by BelKed** [comment]
A similar problem has been reported by `@Dion` on our Discord:

> Watchers have stopped autostarting on linux startup, won't start even through a bash script that runs on WM startup. only manual starting through the terminal works. logs show the same thing for both window and afk watcher:
> 
> ```
> 2023-07-15 12:20:43 [INFO ]: aw-watcher-afk started  (aw_watcher_afk.afk:60)
> 2023-07-15 12:20:44 [INFO ]: Connection to aw-server established by aw-watcher-afk  (aw_client.client:436)
> 2023-07-15 12:20:49 [INFO ]: afkwatcher stopped because parent process died  (aw_watcher_afk.afk:80)
> ```
> 
> aw-qt fully works though

---

This is the solution suggested by `@ErikBjare`:

> Probably because it started before your tray was ready!
> I think someone else has reported an issue with the same problem, and solved it by adding a 5s sleep to the startup script somewhere.
> 
> Ideally, aw-qt would wait itself for x amount of time for a tray to become available, before (optionally) quitting.

**Comment by fernando-gap** [comment]
Hey! There's a error in the script of start.sh, I faced the same issue:

Looking at the code of one of the aw-watchers afk or window, I found out that the function causing problem was the `os.getppid() == 1`. When that condition is true, the watchers throw that exception.

When this function returns 1, this means that if the parent `start.sh` terminated, then all other sub-process killed too!

When notify-send ends, the script would terminate (our parent process), then os.getppid() would return 1, which causes to fail. Interesting enough, because the server is detached from this main process, it isn't affected.

In the documentation the code is:
```./aw-server/aw-server &
./aw-watcher-afk/aw-watcher-afk &
./aw-watcher-window/aw-watcher-window &
notify-send "ActivityWatch started"
```

To solve this issue, the start.sh should be:

```
./aw-watcher-afk/aw-watcher-afk &
./aw-watcher-window/aw-watcher-window &
notify-send "ActivityWatch started" # optional
./aw-server/aw-server;
```

The two process would start in the background, and the server would hold the main process until it's termination, which causes `os.getppid()` to return 1, by that time it would ensure that all three process would end.
